src/Paket.Core/Dependencies/PackageResolver.fs

```diff
@@ -1438,46 +1442,29 @@ let Resolve (getVersionsRaw : PackageVersionsFunc, getPreferredVersionsRaw : Pre
         FirstTrial  = true
         UnlistedSearch = false
     }
+        
+    let rec tryStep result =
+        match result with
+        | StepResult.State state -> state
+        | StepResult.Stage (stage, stackpack, ver, flags) ->
+            tryStep (step stage stackpack ver flags)
 
-    let inline calculate () = step (Step((currentConflict,startingStep,currentRequirement),[])) stackpack Seq.empty flags
+    let inline calculate () =
+        let step = Step((currentConflict, startingStep, currentRequirement), [])
+        tryStep (StepResult.Stage (step, stackpack, Seq.empty, flags))
 
     // Flag to ensure that we don't hide underlying exceptions in the finally block.
     let mutable exceptionThrown = false
     try
-#if DEBUG && !DOTNETCORE
```

Comment:
This code is no longer needed because we do not use recursion in `fuseConflicts` and `step` functions